File: backend_python/app/models/review.py
"""
Review model
"""
from datetime import datetime
from typing import Optional
from beanie import Document, Link
from pydantic import Field
from app.models.user import User
from app.models.service_listing import ServiceListing


class Review(Document):
    """Review model"""
    service_id: str = Field(..., alias="serviceId")
    service: Link[ServiceListing]
    user_id: str = Field(..., alias="userId")
    user: Link[User]
    rating: int
    comment: Optional[str] = None
    created_at: datetime = Field(default_factory=datetime.utcnow, alias="createdAt")
    
    class Settings:
        name = "Review"

File: frontend/lib/domain/entities/review.dart
import 'package:equatable/equatable.dart';

class ReviewEntity extends Equatable {
  const ReviewEntity({
    required this.id,
    required this.serviceId,
    required this.userId,
    required this.rating,
    this.comment,
    this.createdAt,
  });

  final String id;
  final String serviceId;
  final String userId;
  final int rating;
  final String? comment;
  final DateTime? createdAt;

  @override
  List<Object?> get props =>
      [id, serviceId, userId, rating, comment, createdAt];
}
